src/components/Dashboard/index.js

```diff
@@ -131,4 +135,4 @@ const Dashboard = () => {
     );
 };
 
-export default Dashboard;
+export default withErrorAnnouncer(Dashboard);
```

Comment:
Rather than wrapping `Dashboard` and passing `showErrorAnnouncer` down through `DashboardSection` to every item/component, could only `AppItem` be wrapped, since it's the only item that needs this for now?

I haven't tried wrapping a class component yet, but I don't see why it shouldn't work. So the last line of `src/components/Dashboard/DashboardItem/AppItem.js` could be updated to this:
```jsx
export default withErrorAnnouncer(AppItem);
```

Reply:
Ah 👌 
I suspected there was a reason, so thanks for clarifying.

Maybe another way to prevent passing this prop around would be to create a helper component in `AppItem` that wraps `AppFavorite` and the mutation query and all, which could be wrapped in `withErrorAnnouncer`, and used in `AppItem` where `AppFavorite` is used now.

Maybe something like:
```jsx
const AppItemFavoriteIcon = withErrorAnnouncer((props) => {
        const { baseId, app, showErrorAnnouncer } = props;

        const [isFavorite, setIsFavorite] = useState(app.is_favorite);

        const [favorite] = useMutation(appFavorite, {
            onSuccess: () => {
                setIsFavorite(!isFavorite);
            },
            onError: (e) => {
                showErrorAnnouncer(t("favoritesUpdateError", { error: e }), e);
            },
        });

        const onFavoriteClick = () => {
            favorite({
                isFav: !app.is_favorite,
                appId: app.id,
                systemId: app.system_id,
            });
        };

        return (
                    <AppFavorite
                        isFavorite={isFavorite}
                        isExternal={false}
                        onFavoriteClick={onFavoriteClick}
                        baseId={baseId}
                    />
        );
});
```

This is just some copy-paste from this PR and it's not tested, but just to give an idea 😅

Reply:
I could give that a try. 👍